### PR TITLE
LPS-158715 Poshi Automation | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-158715

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path [hunk 1]
```diff
@@ -60,6 +60,11 @@
 	<td>//div[@class='option-selected']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>SPACING_VALUES</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path:64]: Move to MenuItem.path

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 2]
```diff
@@ -35,6 +35,80 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-154530. The page designer can add negative values on spacing from style books."
+	@priority = "4"
+	test AddNegativeValueOnSpacing {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:40]: Missing feature flag

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 3]
```diff
@@ -1511,6 +1585,140 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-154530. View custom values matches in spacing on page editor."
+	@priority = "4"
+	test ViewValuesMatchesInSpacingOnPageEditor {
+		task ("Given a page designer has a content page with default and custom style books") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1591]: Missing feature flag

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 4]
```diff
@@ -35,6 +35,80 @@ definition {
...
+		task ("When the page designer selects a negative value on spacing") {
+			PageEditor.gotoStyle(fragmentName = "Button");
+
+			PageEditor.configureFragment(marginTop = "1");
+		}
+
+		task ("Then the page designer can see negative style on button fragment at edit mode") {
+			PageEditor.viewFragmentStyle(
+				fragmentName = "Button",
+				marginTop = "1");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:93]: 1, -5px

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 5]
```diff
@@ -35,6 +35,80 @@ definition {
...
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Content Page Name",
+				siteName = "Test Site Name");
+		}
+
+		task ("Then the page designer can see negative style on button fragment at view mode") {
+			ContentPages.viewFragmentCommonStyle(
+				fragmentName = "button",
+				styleName = "marginTop",
+				styleValue = "1");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:108]: 1, -5px

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path (hunk 1 above)
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path:64]: SPACING_VALUE

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path [hunk 6]
```diff
@@ -60,6 +60,11 @@
...
+	<td>//div[contains(@id,'dropdown-menu') and contains(@class,'show')]//button[contains(@data-value,'${key_spacingNumber}')]//strong[contains(@class,'text-truncate')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Dropdown.path:65]: //div[contains(@class,'dropdown-menu') and contains(@class,'show')]//button[span[normalize-space(text())='${key_tokenName}']]//strong[contains(@class,'text-truncate')]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 7]
```diff
@@ -1511,6 +1585,140 @@ definition {
...
+		task ("Then the page designer can see custom values on dropdown menu of spacing") {
+			PageEditor.gotoStyle(fragmentName = "Heading");
+
+			Click(
+				ariaLabel = "Margin Top",
+				locator1 = "Button#ANY_WITH_ARIA_LABEL");
+
+			var i = "0";
+
+			for (var spacingValue : list "1em,2em,3em,Auto") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1670]: tokenValue

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 8]
```diff
@@ -1511,6 +1585,140 @@ definition {
...
+			}
+
+			task ("Add unit values on Spacing") {
+				StyleBooks.selectForm(formName = "Spacing");
+
+				StyleBooks.expandSection(sectionName = "Spacing");
+
+				var i = "0";
+
+				for (var spacingValue : list "1em,2em,3em,Auto") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1605]: tokenValue

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase (hunk 2 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:40]: The feature flag was removed on https://github.com/liferay-echo/liferay-portal/pull/9091/commits/bdde05a0eb1c97fcf06259245df08bad47ce48af

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks

Hi @Tim-Cao, PR updated! Could you have a look at? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 9]
```diff
@@ -1511,6 +1590,140 @@ definition {
...
+		task ("When the page designer accesses published content page at view mode") {
+			PageEditor.publish();
+
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Content Page Name",
+				siteName = "Test Site Name");
+		}
+
+		task ("Then the page designer can see auto style on heading fragment at view mode") {
+			ContentPages.viewFragmentCommonStyle(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1720]: Use AssertCSSValue

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 10]
```diff
@@ -1511,6 +1590,140 @@ definition {
...
+		}
+
+		task ("When the page designer selects Auto in spacing") {
+			DropdownMenuItem.click(menuItem = "Spacer 3");
+
+			PageEditor.waitForAutoSave();
+		}
+
+		task ("Then the page designer can see auto style on heading fragment at edit mode") {
+			PageEditor.viewFragmentStyle(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1692]: Use AssertCSSValue

**Comment by ozysouza** [comment]
Hi @Tim-Cao, I've pushed a new commit with the last change.

**Comment by Tim-Cao** [comment]
Good job! LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**